src/main/java/uk/gov/hmcts/reform/sendletter/logging/AppInsights.java

```diff
@@ -34,12 +35,15 @@ public AppInsights(TelemetryClient telemetry) {
     // dependencies
 
     @Pointcut("@annotation(dependency)")
```

Comment:
Do we need to update this `ExternalDependency` or `InternalDependency`?

Reply:
No, the `dependency` is referring to argument name not the type. Unfortunately, cannot unify to use generic case